src/Neo/ProtocolSettings.cs

```diff
@@ -146,11 +146,21 @@ public static ProtocolSettings Load(Stream stream)
         /// <returns>The loaded <see cref="ProtocolSettings"/>.</returns>
         public static ProtocolSettings Load(string path, bool optional = true)
         {
-            using var config = new ConfigurationBuilder().AddJsonFile(path, optional).Build();
-            var section = config.GetSection("ProtocolConfiguration");
-            var settings = Load(section);
-            CheckingHardfork(settings);
-            return settings;
+            if (!File.Exists(path))
+            {
+                if (optional)
```

Comment:
What is the point of optional here? Just use what you had before. Should be renamed to properly identify `optional`, something along the lines of `useDefaultOnNonExists`. Nor even the comments discard this.

Reply:
Preserve the behavior, if it's my decision I remove this arg

Reply:
than remove it.

Reply:
In a different PR